toolkit/offchain/src/reserve/update_settings.rs

```diff
@@ -0,0 +1,233 @@
+		val.set_coin(&min_ada);
+		let a = amount_builder.with_value(&val).build()?;
+		tx_builder.add_output(&a)?;
+	}
+
+	let gov_tx_input = governance.utxo_id_as_tx_input();
+	tx_builder.add_mint_one_script_token_using_reference_script(
+		&governance.policy_script,
+		&gov_tx_input,
+		&governance_script_cost,
+	)?;
+	tx_builder.add_script_reference_input(
+		&reserve.illiquid_circulation_supply_validator_version_utxo.to_csl_tx_input(),
+		reserve.scripts.illiquid_circulation_supply_validator.bytes.len(),
+	);
+	tx_builder.add_required_signer(&ctx.payment_key_hash());
+
+	tx_builder.balance_update_and_build(ctx)
+}
+
+#[cfg(test)]
+mod tests {
+
+	use cardano_serialization_lib::{ExUnits, Language, NetworkIdKind};
+	use hex_literal::hex;
+	use ogmios_client::types::{OgmiosTx, OgmiosUtxo};
+	use partner_chains_plutus_data::reserve::ReserveDatum;
+	use sidechain_domain::{AssetName, PolicyId, TokenId};
+
+	use super::update_reserve_settings_tx;
+	use crate::{
+		csl::TransactionContext,
+		init_governance::GovernanceData,
+		plutus_script::PlutusScript,
+		reserve::ReserveData,
+		scripts_data::ReserveScripts,
+		test_values::{make_utxo, payment_addr, payment_key, protocol_parameters},
+	};
+
+	const REWARDS_TOKEN_POLICY_ID: PolicyId =
+		PolicyId(hex!("1fab25f376bc49a181d03a869ee8eaa3157a3a3d242a619ca7995b2b"));
+
+	// Reward token
+	const REWARDS_TOKEN_ASSET_NAME_STR: &str = "52657761726420746f6b656e";
+
+	#[test]
+	fn update_reserve_settings_tx_test() {
+		let parameters = crate::reserve::create::ReserveParameters {
+			initial_incentive: 100,
+			total_accrued_function_script_hash: PolicyId([233u8; 28]),
+			token: TokenId::AssetId {
+				policy_id: REWARDS_TOKEN_POLICY_ID,
+				asset_name: AssetName::from_hex_unsafe(REWARDS_TOKEN_ASSET_NAME_STR),
+			},
+			initial_deposit: 500000,
+		};
+
+		let reserve: ReserveData = ReserveData {
+			scripts: ReserveScripts {
+				validator: test_validator_script(),
+				auth_policy: test_auth_policy_script(),
+				illiquid_circulation_supply_validator: test_ics_validator_script(),
+			},
+			auth_policy_version_utxo: test_auth_policy_version_utxo(),
+			validator_version_utxo: test_validator_version_utxo(),
+			illiquid_circulation_supply_validator_version_utxo: test_ics_validator_version_utxo(),
+		};
+		let tx = update_reserve_settings_tx(
+			&parameters,
+			&reserve,
+			&test_governance_data(),
+			governance_script_cost(),
+			redeemer_script_cost(),
+			&test_transaction_context(),
+		)
+		.unwrap();
+
+		let body = tx.body();
+		let inputs = body.inputs();
+		let outputs = body.outputs();
+
+		// Both inputs are used to cover transaction
+		assert_eq!(
+			inputs.get(0).to_string(),
+			"1212121212121212121212121212121212121212121212121212121212121212#0"
+		);
+		assert_eq!(
+			inputs.get(1).to_string(),
+			"7474747474747474747474747474747474747474747474747474747474747474#3"
+		);
+		assert_eq!(
+			outputs.get(0).address().to_hex(),
+			"7090fa373007141632b49a920ccccc5bf14d8ed0d2cce51a2354dd423a"
```

Comment:
Cosmetic: could it be compared with `test_validator_script().address(Testnet)` or something like this? I mean, w/o hardcoding value that will not survive if plutus code changes.